### ISSUE TITLE
Coffeescript not compiling

### ISSUE THREAD

**Issue by sargas** [jekyll/jekyll#4675]
###### What version of Jekyll are you using (`jekyll -v`)?
jekyll 3.1.2

###### What operating system are you using?
Gentoo Linux

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, and the command you ran)

When running `jekyll build` or `jekyll serve`, coffeescript files are copied instead of converted. I do have the gem installed and in my _config.yml:

```
$ gem list|egrep -i 'jekyll|coffee'
coffee-script (2.4.1)
coffee-script-source (1.10.0)
jekyll (3.1.2)
jekyll-coffeescript (1.0.1)
jekyll-gist (1.4.0)
jekyll-paginate (1.1.0)
jekyll-sass-converter (1.4.0)
jekyll-watch (1.3.1)
```

```
$ grep gems _config.yml
gems: [jekyll-coffeescript]
```

I believe this started when my distro upgraded from jekyll 2.5.3 to 3.1.2. AFAIK, the only _config change I needed to make was the addition of the `gems` line.

Both of my `.coffee` files start with 
```
---
---
```

###### What did you expect to see?
`.js` files instead of `.coffee` files.

###### What did you see instead?
```
$ jekyll build -V
   Logging at level: debug
Configuration file: /home/joe/dev/hops-site/_config.yml
         Requiring: kramdown
            Source: /home/joe/dev/hops-site
       Destination: /home/joe/dev/hops-site/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       EntryFilter: excluded /README.md
       EntryFilter: excluded /Rakefile
       EntryFilter: excluded /scripts
         Rendering: /assets/themes/js/detail.coffee
  Pre-Render Hooks: /assets/themes/js/detail.coffee
  Rendering Liquid: /assets/themes/js/detail.coffee
  Rendering Markup: /assets/themes/js/detail.coffee
         Rendering: hops_detail.html
  Pre-Render Hooks: hops_detail.html
  Rendering Liquid: hops_detail.html
  Rendering Markup: hops_detail.html
  Rendering Layout: hops_detail.html
         Rendering: index.html
  Pre-Render Hooks: index.html
  Rendering Liquid: index.html
  Rendering Markup: index.html
  Rendering Layout: index.html
         Rendering: /assets/themes/js/main.coffee
  Pre-Render Hooks: /assets/themes/js/main.coffee
  Rendering Liquid: /assets/themes/js/main.coffee
  Rendering Markup: /assets/themes/js/main.coffee
$ ls _site/assets/themes/js
detail.coffee  main.coffee  toggles.min.js
```

As an aside, putting fake output in gems (like `gems: [jekyll-coffeescript, afsadfsdf]`) doesn't give any errors, so I don't know if it is correctly finding the jekyll-coffeescript gem or not.
I also tried patching jekyll-coffeescript-1.0.1 with https://github.com/jekyll/jekyll-coffeescript/compare/2ab141bfa4c62a0a942360e8a9bc824f6d7f9a24...master.diff (to effectively use the current git version) but that hasn't affected jekyll.

**Comment by sargas** [comment]
I should add that I'd be happy to submit this to jekyll/jekyll-coffeescript instead, I just don't have any indication that jekyll is even using the gem.

**Comment by parkr** [comment]
@sargas Do you have any extraneous spaces after the first set of `---`?

**Comment by sargas** [comment]
Nope.
```
cat -E assets/js/test.coffee
--$
--$
$
```
```
cat -E _site/assets/js/test.coffee
--$
--$
$
```

**Comment by envygeeks** [comment]
@sargas you are missing a dash, you only have two dashes, we require 3 for front-matter.

**Comment by sargas** [comment]
@envygeeks Sorry about that. Jekyll is reading the front-matter, just doesn't seem to be passing it to coffeescript.
```
cat -E assets/js/test.coffee
---$
---$
"Invalid " -> "Coffeescript"$
```
```
cat -E _site/assets/js/test.coffee
"Invalid " -> "Coffeescript"$
```

**Comment by envygeeks** [comment]
@sargas add an extra line-space between the last `---` and the source and see if that fixes it.

**Comment by sargas** [comment]
No, although jekyll seems to eat that extra newline:
```
cat -E assets/js/test.coffee
---$
---$
$
"Invalid " -> "Coffeescript" -> "Test2"$
```
```
cat -E _site/assets/js/test.coffee
"Invalid " -> "Coffeescript" -> "Test2"$
```

**Comment by envygeeks** [comment]
Is there anyway you can upload an example repository I can have my bot shim and trace?

**Comment by sargas** [comment]
Sorry about all this, sometime when I was messing with this a month ago I removed the gems line, and didn't realize I was missing it when trying the steps above.

With `gems: [jekyll-coffeescript]` in `_config.yaml` the coffeescript is successfully rendered, and nonsense in that line does throw an error. I'm not sure what changed in the last month to get it working, although I have upgraded rubygems since then.

Thanks!
